Tasks/AzureRmWebAppDeploymentV3/Tests/L0.ts

```diff
@@ -57,25 +58,39 @@ describe('AzureRmWebAppDeployment Suite', function() {
     }
 
     it('Runs successfully with XML variable substitution', (done:MochaDone) => {
-        let tp = path.join(__dirname, "..", "webdeployment-common", "Tests", 'L1XmlVarSub.js');
-        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        let tp = path.join(__dirname, "..", "node_modules", "azure-pipelines-tasks-webdeployment-common", "Tests", 'L1XmlVarSub.js');
+        let tr = new ttm.MockTestRunner(tp);
         tr.run();
-
-        var resultFile = ltx.parse(fs.readFileSync(path.join(__dirname,  "..", "webdeployment-common","Tests", 'L1XmlVarSub', 'Web_test.config')));
-        var expectFile = ltx.parse(fs.readFileSync(path.join(__dirname, "..", "webdeployment-common","Tests", 'L1XmlVarSub', 'Web_Expected.config')));
-        assert(ltx.equal(resultFile, expectFile) , 'Should have substituted variables in Web.config file');
-        var resultFile = ltx.parse(fs.readFileSync(path.join(__dirname, "..", "webdeployment-common", "Tests", 'L1XmlVarSub', 'Web_test.Debug.config')));
-        var expectFile = ltx.parse(fs.readFileSync(path.join(__dirname, "..", "webdeployment-common", "Tests", 'L1XmlVarSub', 'Web_Expected.Debug.config')));
-        assert(ltx.equal(resultFile, expectFile) , 'Should have substituted variables in Web.Debug.config file');   
-        var resultParamFile = ltx.parse(fs.readFileSync(path.join(__dirname,  "..", "webdeployment-common","Tests", 'L1XmlVarSub', 'parameters_test.xml')));
-        var expectParamFile = ltx.parse(fs.readFileSync(path.join(__dirname,  "..", "webdeployment-common","Tests", 'L1XmlVarSub', 'parameters_Expected.xml')));
-        assert(ltx.equal(resultParamFile, expectParamFile) , 'Should have substituted variables in parameters.xml file');
-
+        let transformedFilePath = path.join(__dirname, "..", "node_modules", "azure-pipelines-tasks-webdeployment-common", "Tests", 'L1XmlVarSub', 'Web_test.config');
+        let expectedFilePath = path.join(__dirname, "..", "node_modules", "azure-pipelines-tasks-webdeployment-common", "Tests", 'L1XmlVarSub', 'Web_Expected.config');
+        let transformedFileAsBuffer = fs.readFileSync(transformedFilePath);
+        let expectedFileAsBuffer = fs.readFileSync(expectedFilePath);
+        let transformedFileEncodeType = fileEncoding.detectFileEncoding(transformedFilePath, transformedFileAsBuffer)[0];
+        let transformedFileAsString = transformedFileAsBuffer.toString(transformedFileEncodeType);
+        transformedFileAsString = transformedFileAsString.replace( /[\n]+/gm, "\r\n" );
```

Comment:
What is the purpose of this line? Is it to ensure that all line endings are `\r\n\`? But what if the line ending is already `\r\n`, wouldn't this create a sequence of `\r\r\n`?

Reply:
But it seems like it would be a simple enough to fix. ASfter the first `.replace`, just run  `.replace` again to replace `\r\r` with `\r`? (I'm sure it's possible to do with a single regex, but I'm not a regex wizard :))

Reply:
Done, thank you.